Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,779 @@
+    var scrollLeft = scrollableContainer.scrollLeft;
+    var previousScrollLeft = scrollableContainerJQuery.data("bocListPreviousScrollLeft");
+    var fakeTableHeadScrollLeft = fakeTableHeadContainer.scrollLeft;
+
+    var hasScrollMoveFromScrollbar = previousScrollLeft !== scrollLeft;
+    var hasScrollMoveFromColumnHeader = previousScrollLeft !== fakeTableHeadScrollLeft;
+    if (!hasScrollMoveFromScrollbar && !hasScrollMoveFromColumnHeader)
+      return;
+
+    scrollLeft = hasScrollMoveFromColumnHeader ? fakeTableHeadScrollLeft : scrollLeft;
+    scrollableContainerJQuery.data("bocListPreviousScrollLeft", scrollLeft);
+
+    if (hasScrollMoveFromColumnHeader)
+      scrollableContainer.scrollLeft = scrollLeft;
+
+    if (hasScrollMoveFromScrollbar)
+      fakeTableHeadContainer.scrollLeft = scrollLeft;
+  }
+
+  public static InitializeNavigationBlock (pageNumberField: JQuery, pageIndexField: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsObject ('pageNumberField', pageNumberField);
+    ArgumentUtility.CheckNotNullAndTypeIsObject('pageIndexField', pageIndexField);
+  
+    pageNumberField.bind('change', function () {
+  
+      var pageNumber = parseInt(pageNumberField.val(), 10);
+      if (isNaN (pageNumber) || !TypeUtility.IsInteger (pageNumber))
+      {
+        if (pageNumberField.val().length > 0)
+          setTimeout(function () { pageNumberField.focus(); }, 0);
+  
+        pageNumberField.val(String(parseInt(pageIndexField.val(), 10) + 1));
+        return false;
+      }
+      else
+      {
+        pageIndexField.val (String(pageNumber - 1));
+        pageIndexField.trigger('change');
+        return true;
+      }
+    });
+  
+    pageNumberField.bind('keydown', function (event) {
+      var enterKey = 13;
+      var zeroKey = 48;
+      var nineKey = 57;
+      var zeroKeyNumBlock = 96;
+      var nineKeyNumBlock = 105;
+      var f1Key = 112;
+      var f12Key = 123;
+      var isEnterKey = event.keyCode == enterKey;
+      var isControlKey = event.keyCode < zeroKey || event.keyCode >= f1Key && event.keyCode <= f12Key;
+      var isNumericKey = event.keyCode >= zeroKey && event.keyCode <= nineKey || event.keyCode >= zeroKeyNumBlock && event.keyCode <= nineKeyNumBlock;
+  
+      if (isEnterKey)
+      {
+        pageNumberField.trigger("change");
+        event.cancelBubble = true;
+        event.stopPropagation();
+        return false;
+      }
+      else if (event.altKey || event.ctrlKey || isControlKey || isNumericKey)
+      {
+        return true;
+      }
+      else
+      {
+        return false;
+      }
+    });
+  }
+}
+
+function BocList_InitializeGlobals(): void
+{
+  BocList.InitializeGlobals();
+}
+
+function BocList_InitializeList(bocList: HTMLElement, selectRowSelectorControlName: string, selectAllSelectorControlName: string, selection: Dictionary<HTMLElement> | number, hasClickSensitiveRows: boolean, onSelectionChangedHandler: { (boclist: HTMLElement, flag: boolean): void }): void
```

Comment:
The `selection` parameter should be `number` if I am not mistaken. This should then also be changed in the `BocList_SelectedRows` class.

Reply:
@a-ctor The thing is, in the JavaScript version, there is a comment right below `selection` that statates
```
//  Associative Array: <SelectorControl ID>, <BocList_RowBlock>
```
However, numbers are being assigned to it as well, which has me confused. Do you think I should still just change it to a number?

Reply:
If it works with just the `number` type we should remove the `Dictionary`. From my quick glance it is only compared to numbers and initialized to numbers. The comment might be outdated and probably can be removed. You can blame the comment to make sure.